src/stream/src/executor/hash_join.rs

```diff
@@ -52,6 +52,10 @@ use crate::task::AtomicU64Ref;
 /// enum is not supported in const generic.
 // TODO: Use enum to replace this once [feature(adt_const_params)](https://github.com/rust-lang/rust/issues/95174) get completed.
 pub type JoinTypePrimitive = u8;
+
+/// Evict the cache every n rows.
```

Comment:
Should we do this according to the input rows or output rows? 🤔

Reply:
There is really no perfect choice for this. We just hope this change can mitigate the issue instead of fully resolving that. 

Input row works here because the cache entry is 1-1 for every input row. If in some cases the OOM still happens, we should consider decreasing the EVICT_EVERY_N_ROWS (even to 1).

Reply:
Actually I think do eviction on every Chunk is good enough 🥵

Reply:
One of the motivations for EVICT_EVERY_N_ROWS is to decouple the eviction time with chunk size.